Modules/_datetimemodule.c

```diff
@@ -6491,36 +6472,36 @@ PyInit__datetime(void)
      * values. This may change in the future.*/
     delta = new_delta(-1, 60, 0, 1); /* -23:59 */
     if (delta == NULL)
-        return NULL;
+        return -1;
     x = create_timezone(delta, NULL);
     Py_DECREF(delta);
     if (x == NULL || PyDict_SetItemString(d, "min", x) < 0)
-        return NULL;
+        return -1;
     Py_DECREF(x);
 
     delta = new_delta(0, (23 * 60 + 59) * 60, 0, 0); /* +23:59 */
     if (delta == NULL)
-        return NULL;
+        return -1;
     x = create_timezone(delta, NULL);
     Py_DECREF(delta);
     if (x == NULL || PyDict_SetItemString(d, "max", x) < 0)
-        return NULL;
+        return -1;
     Py_DECREF(x);
 
     /* Epoch */
     PyDateTime_Epoch = new_datetime(1970, 1, 1, 0, 0, 0, 0,
                                     PyDateTime_TimeZone_UTC, 0);
     if (PyDateTime_Epoch == NULL)
-      return NULL;
+      return -1;
 
     /* module initialization */
-    PyModule_AddIntMacro(m, MINYEAR);
-    PyModule_AddIntMacro(m, MAXYEAR);
+    PyModule_AddIntMacro(module, MINYEAR);
+    PyModule_AddIntMacro(module, MAXYEAR);
 
     x = PyCapsule_New(&CAPI, PyDateTime_CAPSULE_NAME, NULL);
```

Comment:
I don't think that it's correct to override the capsule if PyInit__datetime() is called twice.

Reply:
CAPI is not fully static, there is a heap-allocated type that is created on exec. Not sure we can change this right now.

https://github.com/python/cpython/pull/19122/commits/9a424288ed61cb93a5e0787a6cf421b02d226ffa#diff-0090febf24974047330d8a02d4dbdc79L6486-L6487

Reply:
Why can't we change it?

If it's important that we have two separate `PyCapsule_New` objects, doesn't that mean that if you have a C extension running in two subinterpreters that references this capsule, they'll get a different value for `PyDateTimeAPI` depending on when they look up the reference rather than depending on which subinterpreter they are running from?

Not sure I see a way around this without a breaking change to the API.